### ISSUE TITLE
Receiving integer enum Query Parameters as String

### ISSUE THREAD

**Issue by joanTrave** [tiangolo/fastapi#5047]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional, Any

from enum import Enum

from fastapi import FastAPI, Depends
from pydantic import BaseModel

app: FastAPI = FastAPI()


class MyEnum(int, Enum):
    STATE_0 = 0
    STATE_1 = 1


class MyCommons(BaseModel):
    state: Optional[MyEnum] = None


@app.get("/")
def get_things(commons: MyCommons = Depends(MyCommons)) -> Any:
    ...
```

### Description

* Open the browser and call the endpoint /?state=1
* I expect to get 200 OK with null payload
* I receive 422 Unprocessable Entity with payload:
```{
  "detail": [
    {
      "loc": [
        "query",
        "state"
      ],
      "msg": "value is not a valid enumeration member; permitted: 0, 1",
      "type": "type_error.enum",
      "ctx": {
        "enum_values": [
          0,
          1
        ]
      }
    }
  ]
}
```
* If I change MyEnum implementation to:
```
class MyEnum(int, Enum):
    STATE_0 = 0
    STATE_1 = 1
```
* I receive 200 OK with null payload response as expected.

### Operating System

macOS

### FastAPI Version

0.78.0

### Python Version

3.8.9

**Comment by JarroVGIT** [comment]
There is a special class for this, called `IntEnum`. The following code runs as is;

```python
from typing import Optional, Any
from enum import Enum, IntEnum
from fastapi import FastAPI, Depends
from pydantic import BaseModel

app: FastAPI = FastAPI()

class MyEnum(IntEnum):
    STATE_0 = 0
    STATE_1 = 1

class MyCommons(BaseModel):
    state: Optional[MyEnum] = None

@app.get("/")
def get_things(commons: MyCommons = Depends(MyCommons)) -> Any:
    return commons

if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8000, )   
```

**Comment by joanTrave** [comment]
Thanks for the fast response!
I have checked if there is any reference to this class in the official documentation and have not found any reference.

**Comment by nadavamami** [comment]
you can look at [https://docs.python.org/3/library/enum.html] IntEnum enforce members are integers and members can be compared with integers

**Comment by joanTrave** [comment]
I agree with you. Nevertheless, the same functionality can be achieved with a subclass implementation between the `int` and `enum` types. This ambiguity should be cleared up in the library documentation.
